sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/FileStreamSource.scala

```diff
@@ -67,12 +62,33 @@ class FileStreamSource(
   }
 
   /**
+   * Set the metadata path. This method should be called before using [[FileStreamSource]].
+   */
+  def setMetadataPath(metadataPath: String): Unit = {
```

Comment:
Sure, but if you find yourself hacking around the fact that we don't know some information at some point in the control flow and its making the implementation a lot more complicated, then we need to rethink the control flow.